src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/EnchantmentRune.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * @author Linox
+ *
+ * @see ItemDropHandler
+ * @see Enchantment
+ *
+ */
+public class EnchantmentRune extends SimpleSlimefunItem<ItemDropHandler> {
+
+    private static final double RANGE = 1.5;
+    private final HashMap<Material, Enchantment[]> applicableEnchs = new HashMap<>();
+
+    public EnchantmentRune(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+
+        for (Material mat : Material.values()) {
+            Set<Enchantment> enchSet = new HashSet<>();
+            for (Enchantment ench : Enchantment.values()) {
+                if (ench.canEnchantItem(new ItemStack(mat))) enchSet.add(ench);
+            }
+            applicableEnchs.put(mat, enchSet.toArray(new Enchantment[0]));
+        }
+    }
+
+    @Override
+    public ItemDropHandler getItemHandler() {
+        return (e, p, item) -> {
+            if (isItem(item.getItemStack())) {
+
+                if (!Slimefun.hasUnlocked(p, SlimefunItems.ENCHANTMENT_RUNE, true)) {
+                    return true;
+                }
+
+                Slimefun.runSync(() -> activate(p, e, item), 20L);
+
+                return true;
+            }
+            return false;
+        };
+    }
+
+    private void activate(Player p, PlayerDropItemEvent e, Item item) {
+        // Being sure the entity is still valid and not picked up or whatsoever.
+        if (!item.isValid()) {
+            return;
+        }
+
+        Location l = item.getLocation();
+        Collection<Entity> entites = l.getWorld().getNearbyEntities(l, RANGE, RANGE, RANGE, this::findCompatibleItem);
+        Optional<Entity> optional = entites.stream().findFirst();
+
+        if (optional.isPresent()) {
+            Item entity = (Item) optional.get();
+            ItemStack target = entity.getItemStack();
+
+            Enchantment[] enchArr = findEnchArr(target.getType());
+            if (enchArr.length == 0) return;
+            int enchIndex = ThreadLocalRandom.current().nextInt(enchArr.length);
+            Enchantment ench = enchArr[enchIndex];
+
+            int level = 1;
+            if (ench.getMaxLevel() != 1) level = ThreadLocalRandom.current().nextInt(ench.getMaxLevel() + 1);
+
+            target.addEnchantment(ench, level);
+
+            if (target.getAmount() == 1) {
+                e.setCancelled(true);
+
+                // This lightning is just an effect, it deals no damage.
+                l.getWorld().strikeLightningEffect(l);
+
+                Slimefun.runSync(() -> {
+                    // Being sure entities are still valid and not picked up or whatsoever.
+                    if (item.isValid() && entity.isValid() && target.getAmount() == 1) {
+
+                        l.getWorld().createExplosion(l, 0);
+                        l.getWorld().playSound(l, Sound.ENTITY_GENERIC_EXPLODE, 0.3F, 1);
```

Comment:
Should probably choose a little more magical sound effect and visual than a quote: *GENERIC_SPLODE*...